azurerm/internal/services/synapse/synapse_private_link_hub_test.go

```diff
@@ -0,0 +1,116 @@
+package synapse_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/synapse/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+type SynapsePrivateLinkHubResource struct{}
+
+func TestAccSynapsePrivateLinkHub_create(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_private_link_hub", "test")
+	r := SynapsePrivateLinkHubResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+	})
+}
+
+func TestAccSynapsePrivateLinkHub_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_private_link_hub", "test")
+	r := SynapsePrivateLinkHubResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config: r.withUpdateFields(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("tags.ENV").HasValue("Test2"),
+			),
+		},
+	})
+}
+
+func (r SynapsePrivateLinkHubResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.PrivateLinkHubID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Synapse.PrivateLinkHubsClient.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving Synapse Private Link Hub %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r SynapsePrivateLinkHubResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_private_link_hub" "test" {
+  name                = "acctestsw%d"
```

Comment:
Using indexed verbs will make it easier to read and reduce repetition if config needs to be added at a later date
```suggestion
%[1]s

resource "azurerm_synapse_private_link_hub" "test" {
  name                = "acctestsw%[2]d"
```